drivers/timer/arcv2_timer0.c

```diff
@@ -205,10 +205,14 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 
 	/* Round delay up to next tick boundary */
 	delay = ((delay + CYC_PER_TICK - 1) / CYC_PER_TICK) * CYC_PER_TICK;
-	last_load = delay;
 
-	timer0_limit_register_set(last_load - 1);
-	timer0_control_register_set(_ARC_V2_TMR_CTRL_NH | _ARC_V2_TMR_CTRL_IE);
+	if (last_load != delay) {
+		timer0_count_register_set(elapsed());
```

Comment:
not sure I like the call to 'elapsed()' here, I'd rather get rid of elapsed later and use another timer/counter for getting a cycle count without overflow or timeout wrapping problems.
Just look at the timer control bit to see if it expired already?

Reply:
There's also the problem that writing the counter to a value that depends on reading the running counter causes some cycles to be "lost".  Doing this too much will slowly degrade the accuracy/phase of the tick arrivals.